### PR TITLE
Added docs for controlling widget visibility

### PR DESCRIPTION
**Pull request by Rithvik-Reddy**
PR for issue:  https://github.com/appsmithorg/appsmith/issues/1312
The commit adds an extra page to docs to show users how to control the visibility of widgets dynamically.

### CODE REVIEW

**Review by Nikhil-Nandagopal — CHANGES_REQUESTED** [review]
@Rithvik-Reddy I've added my comments

File: core-concepts/building-the-ui/widget-visibility.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+description: >-
+  The visibility of widgets can be controlled dynamically through another widget or through an API.
```
[the remark this answers is not in the document]
Comment (Rithvik-Reddy) [reply core-concepts/building-the-ui/widget-visibility.md:3]: Alright

File: core-concepts/building-the-ui/widget-visibility.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+We can use the value obtained from an API to dynamically set the visibility of a widget using the JS mode for the visible option. 
+
+![Click to expand](../../.gitbook/assets/widget-visibility-api.gif)
+
+In the example shown above the value of the radio buttons is sent as a parameter to the API named "getObject". The API response is then used to dynamically change the visibility of a widget using a condition. 
+
+The condition below is used in object A
+
+```text
```
[the remark this answers is not in the document]
Comment (Rithvik-Reddy) [reply core-concepts/building-the-ui/widget-visibility.md:28]: The GIF that I added shows how to control the visibility of a widget from another widget dynamically. Do you want me to add the an example of the dyanmic forms?

**Comment by Rithvik-Reddy** [comment]
@Nikhil-Nandagopal I commented a few questions, please let me know before I make changes.

File: core-concepts/building-the-ui/widget-visibility.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nikhil-Nandagopal) [reply core-concepts/building-the-ui/widget-visibility.md:28]: @Rithvik-Reddy yes as an additional example, show a dynamic form using a tab widget to hide / show different flows

[the remark this answers is not in the document]
Comment (Rithvik-Reddy) [reply core-concepts/building-the-ui/widget-visibility.md:28]: Alright

**Comment by Rithvik-Reddy** [comment]
Hi @Nikhil-Nandagopal, Ive committed the requested changes.

**Review by Nikhil-Nandagopal — APPROVED** [review]

**Comment by Nikhil-Nandagopal** [comment]
@Rithvik-Reddy Thank you for the contribution 🎉

**Pull request merged**